src/lib/services/nft-collection/api.ts

```diff
@@ -0,0 +1,88 @@
+import axios from "axios";
+
+import {
+  zActivitiesResponse,
+  zCollectionByCollectionAddressResponse,
+  zCollectionCreatorResponse,
+  zCollectionMutateEventsResponse,
+  zCollectionsByAccountAddressResponse,
+  zNftCollectionsResponse,
+} from "../types";
+import type { HexAddr, HexAddr32 } from "lib/types";
+import { parseWithError } from "lib/utils";
+
+export const getNftCollections = async (
+  endpoint: string,
+  limit: number,
+  offset: number,
+  search = ""
+) =>
+  axios
+    .get(`${endpoint}`, {
+      params: {
+        limit,
+        offset,
+        search,
+      },
+    })
+    .then(({ data }) => parseWithError(zNftCollectionsResponse, data));
+
+export const getNftCollectionByCollectionAddress = async (
```

Comment:
@songwongtp What do you prefer: `getNftCollection` or `getCollection`?

Reply:
let's go with `getNftCollection`